File: backend/models/ai_stats.py
from pydantic import BaseModel, Field


class TopModelItem(BaseModel):
    provider: str
    model: str
    count: int


class TopProviderItem(BaseModel):
    provider: str
    count: int


class FailureReasonItem(BaseModel):
    reason: str
    count: int


class HourlyItem(BaseModel):
    hour: str
    count: int


class ModelRankingItem(BaseModel):
    provider: str
    model: str
    total: int
    success_rate: float
    avg_latency: float


class ProviderRankingItem(BaseModel):
    provider: str
    total: int
    success_rate: float
    avg_latency: float


class ProviderFailureItem(BaseModel):
    provider: str
    count: int


class AiStatsOverview(BaseModel):
    today_total: int = Field(..., description="今日调用总次数")
    today_success: int = Field(..., description="今日成功次数")
    today_failure: int = Field(..., description="今日失败次数")
    success_rate: float = Field(..., description="今日成功率(%)")
    failure_rate: float = Field(..., description="今日失败率(%)")
    consecutive_failures: int = Field(..., description="当前连续失败次数")
    last_success_time: str | None = Field(None, description="最后一次成功时间")
    avg_latency_ms: float = Field(..., description="今日平均延迟(ms)")
    p50_latency_ms: float = Field(..., description="P50延迟(ms)")
    p95_latency_ms: float = Field(..., description="P95延迟(ms)")
    p99_latency_ms: float = Field(..., description="P99延迟(ms)")
    timeout_rate: float = Field(..., description="超时占比(%)")
    total_all: int = Field(..., description="累计总调用次数")
    cache_hit_count: int = Field(..., description="今日缓存命中次数")
    cache_hit_rate: float = Field(..., description="今日缓存命中率(%)")
    top_model: TopModelItem | None = None
    top_provider: TopProviderItem | None = None
    top_failures: list[FailureReasonItem] = Field(default_factory=list)
    hourly_distribution: list[HourlyItem] = Field(default_factory=list)
    model_ranking: list[ModelRankingItem] = Field(default_factory=list)
    provider_ranking: list[ProviderRankingItem] = Field(default_factory=list)
    provider_failures: list[ProviderFailureItem] = Field(default_factory=list)


class DailyTrendItem(BaseModel):
    date: str
    total: int
    success_count: int
    success_rate: float
    avg_latency: float


class AiCallLogItem(BaseModel):
    id: int
    provider_name: str
    model_name: str
    call_time: str
    success: bool
    latency_ms: int | None
    error_reason: str | None
    from_cache: bool
    triggered_alerts: str | None

File: frontend/src/api/modules/ai-stats.ts
import request from '@/api/request'

export interface AiStatsOverview {
  today_total: number
  today_success: number
  today_failure: number
  success_rate: number
  failure_rate: number
  consecutive_failures: number
  last_success_time: string | null
  avg_latency_ms: number
  p50_latency_ms: number
  p95_latency_ms: number
  p99_latency_ms: number
  timeout_rate: number
  total_all: number
  cache_hit_count: number
  cache_hit_rate: number
  top_model: { provider: string; model: string; count: number } | null
  top_provider: { provider: string; count: number } | null
  top_failures: { reason: string; count: number }[]
  hourly_distribution: { hour: string; count: number }[]
  model_ranking: {
    provider: string
    model: string
    total: number
    success_rate: number
    avg_latency: number
  }[]
  provider_ranking: { provider: string; total: number; success_rate: number; avg_latency: number }[]
  provider_failures: { provider: string; count: number }[]
}

export interface DailyTrendItem {
  date: string
  hour: string | null
  total: number
  success_count: number
  success_rate: number
  avg_latency: number
}

export interface AiCallLogItem {
  id: number
  provider_name: string
  model_name: string
  call_time: string
  success: boolean
  latency_ms: number | null
  error_reason: string | null
  from_cache: boolean
  triggered_alerts: string | null
  raw_response: string | null
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
}

export interface AiCallLogsResponse {
  items: AiCallLogItem[]
  total: number
  page: number
  page_size: number
}

export interface TokenOverview {
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  calls: number
  estimated_cost: number
}

export interface TokenByModel {
  provider_name: string
  model_name: string
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  calls: number
  estimated_cost: number
}

export interface TokenTrendItem {
  date: string
  hour: string | null
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  calls: number
  estimated_cost: number
}

export interface DateParams {
  start_date: string
  end_date: string
}

export const aiStatsApi = {
  getOverview(params: DateParams) {
    return request.get<AiStatsOverview>('/ai-stats/overview', { params })
  },
  getTrend(params: DateParams) {
    return request.get<DailyTrendItem[]>('/ai-stats/trend', { params })
  },
  getLogs(page = 1, pageSize = 20, dateParams?: DateParams) {
    return request.get<AiCallLogsResponse>('/ai-stats/logs', {
      params: { page, page_size: pageSize, ...dateParams },
    })
  },
  getTokenOverview(params: DateParams) {
    return request.get<TokenOverview>('/ai-stats/tokens/overview', { params })
  },
  getTokenByModel(params: DateParams) {
    return request.get<TokenByModel[]>('/ai-stats/tokens/by-model', { params })
  },
  getTokenTrend(params: DateParams) {
    return request.get<TokenTrendItem[]>('/ai-stats/tokens/trend', { params })
  },
}
